modules/fedimint-ln-client/src/incoming.rs

```diff
@@ -225,32 +225,24 @@ impl DecryptingPreimageState {
         global_context: DynGlobalClientContext,
         contract_id: ContractId,
     ) -> Result<Preimage, IncomingSmError> {
-        // TODO: Get rid of polling
-        let preimage = loop {
-            let contract = global_context
+        loop {
```

Comment:
Thought it's this loop we need to rid of? Same comment for the other polls below

Reply:
So, theses loops are harder to get rid of, because the API call can still return a `FederationError` (i.e the federation is offline). If the federation is offline, we don't necessarily want to transition to a failed state, because the federation could come back later and the decryption could be successful. This means that we do want to retry here until we get a clear signal if the decryption was successful.

The improvement here is that we don't poll unless we receive a `FederationError`. `wait_preimage_decryption` will block until there is a clear failure or if the preimage is successfully decrypted. Previously we were polling even if the decryption was in progress.

Same goes for the other API calls.

Reply:
ack.
Pinging @justinmoon, @dpc for further reviews